src/coreimage.cs

```diff
@@ -4729,70 +4733,76 @@ interface CIAttributedTextImageGenerator {
 	[TV (11,0)]
 	[BaseType (typeof (CIFilter))]
 	interface CIBarcodeGenerator {
-		// TODO: Needs review
-		//[CoreImageProperty ("inputBarcodeDescriptor")]
-		//CIBarcodeDescriptor BarcodeDescriptor { get; set; }
+		[CoreImageFilterProperty ("inputBarcodeDescriptor")]
+		CIBarcodeDescriptor BarcodeDescriptor { get; set; }
 	}
 
 	[CoreImageFilter]
 	[iOS (11,0)]
 	[Mac (10,13)]
 	[TV (11,0)]
-	[BaseType (typeof (CIFilter))]
+	[BaseType (typeof (CIFilter))] // Maybe 'CIScaleTransform' (shared 'Scale' and 'AspectRatio' property)?
 	interface CIBicubicScaleTransform {
-		// TODO: Needs review
-		//[CoreImageProperty ("inputB")]
-		//// TODO: this was an NSNumber transformed to float, but maybe an int or bool is more appropriate
-		//float B { get; set; }
-		//[CoreImageProperty ("inputScale")]
-		//// TODO: this was an NSNumber transformed to float, but maybe an int or bool is more appropriate
-		//float Scale { get; set; }
-		//[CoreImageProperty ("inputC")]
-		//// TODO: this was an NSNumber transformed to float, but maybe an int or bool is more appropriate
-		//float C { get; set; }
-		//[CoreImageProperty ("inputAspectRatio")]
-		//// TODO: this was an NSNumber transformed to float, but maybe an int or bool is more appropriate
-		//float AspectRatio { get; set; }
+		[CoreImageFilterProperty ("inputB")]
+		float B { get; set; }
+
+		[CoreImageFilterProperty ("inputC")]
+		float C { get; set; }
+
+		[CoreImageFilterProperty ("inputScale")]
+		float Scale { get; set; }
+
+		[CoreImageFilterProperty ("inputAspectRatio")]
+		float AspectRatio { get; set; }
 	}
 
 	[CoreImageFilter]
+	[Abstract]
 	[iOS (11,0)]
 	[Mac (10,13)]
 	[TV (11,0)]
 	[BaseType (typeof (CIFilter))]
+	interface CIBlur { // `CIMotionBlur.SuperClass == CILinearBlur` but it's marked as `[NoiOS]` and `[NoMac]` so we have ours
+		[CoreImageFilterProperty ("inputRadius")]
+		float Radius { get; set; }
+	}
+
+	[CoreImageFilter]
+	[iOS (11,0)]
+	[Mac (10,13)]
+	[TV (11,0)]
+	[BaseType (typeof (CIBlur))] // `CIBokehBlur` doesn't explicitly have `SuperClass` but it should be `CIBlur` to inherit the `inputRadius` property
```

Comment:
same (about `CIBlur`)
note that adding a base type (in the future) would not be a breaking change if the flattened API is identical

Reply:
not if it's virtual

Reply:
If what is virtual? Nothing's virtual in `CIBokehBlur.g.cs` or `CIBlur.g.cs`

Reply:
Allowed [1]
* _Introducing a new base class_
* _Moving a member onto a class higher in the hierarchy tree of the type from which it was removed_

[1] https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md